CONTRIBUTING.md

```diff
@@ -69,3 +69,87 @@ header.
+A `.out` file **must** exist, and RGBLINK's output must match that file's contents.
+
+Additionally, if a `.out.bin` file exists, the `.gb` file generated by RGBLINK must match it.
+
+#### Linker script tests
+
+These allow applying various linker scripts to the same object file.
+If one or more `.link` files exist, whose names start the same as the `.asm` file, then each of those files correspond to one test.
+
+Each `.link` linker script **must** be accompanied by a `.out` file, and RGBLINK's output must match that file's contents when passed the corresponding linker script.
+
+#### Variant tests
+
+These allow testing RGBLINK's `-d`, `-t`, and `-w` flags.
+If one or more files called <code><var>&lt;asm file name&gt;</var>-<var>&lt;flag&gt;</var>.out</code> or <code><var>&lt;asm file name&gt;</var>-no-<var>&lt;flag&gt;</var>.out</code> exist, then each of them corresponds to one test.
+
+The object file will be linked with and without said flag, respectively; and in each case, RGBLINK's output must match the `.out` file's contents.
+
+### RGBFIX
+
+Each `.bin` file corresponds to one test, and **must** be accompanied by a `.flags` file and a `.err` file.
+
+The `.flags` file is a text file whose first line contains flags to pass to RGBFIX.
+(There may be more lines, which will be ignored; they can serve as commentsto explain what the test is about.)
+
+RGBFIX will be invoked on the `.bin` file, and its error output must match the contents of the `.err` file.
+(If no errors ought to be printed, then the `.err` file should just be empty.)
+
+Additionally, if a `.gb` file exists, the output of RGBFIX must match the `.gb`.
+
+### RGBGFX
+
+Each `.png` file corresponds to one test.
+RGBGFX will be invoked on the file.
+If a `.flags` file exists, it will be used as part of the RGBGFX invocation (<code>@<var>&lt;file&gt;</var>.flags</code>).
+
+If no `.err` file exists, RGBGFX is simply expected to be able to process the file normally.
+If one *does* exist, RGBGFX's return status is ignored, but its output **must** match the `.err` file's contents.
+
+### Downstream projects
+
+1. Make sure the downstream project supports <code>make <var>&lt;target&gt;</var> RGBDS=<var>&lt;path to RGBDS&gt;</var></code>.
+   While the test suite supports any Make target name, only [Make](//gnu.org/software/make) is currently supported, and the Makefile must support a `RGBDS` variable to use a non-system RGBDS.
+
+   Only projects hosted on GitHub are currently supported, too.
+2. Add the project to `test/fetch-test-deps.sh`: add a new `action` line at the bottom, following the existing pattern:
+   
+   ```sh
+   action <owner>/<repo>   <date of last commit>   <hash of last commit>
+   ```
+
+   (The date is used to avoid fetching too much history when cloning the repositories.)
+3. Add the project to `test/run-tests.sh`: add a new `test_downstream` line at the bottom, following the existing pattern:
+
+   ```sh
+   test_downstream <repo>  <Make target>
```

Comment:
```suggestion
   test_downstream  <repo>  <make target>
```

Reply:
Perhaps
```suggestion
   test_downstream <repo>  <makefile target>
```
then?

Reply:
I thought it was "`make` target" as in "target of the `make` command", but sure.

Reply:
Correct, but without the extra formatting, I don't think it's clear enough that this is referring to the command and not the English verb. (Hence my original capitalisation, to try drawing attention to that.)

Since the targets are defined in the Makefile, I think both formulations are acceptable, but "makefile target" is less ambiguous.

Reply:
Whichever you prefer, then.